### ISSUE TITLE
Plugins

### ISSUE THREAD

**Issue by madmediaz2** [mschnitzer/ark-survival-ascended-linux-container-image#66]
is the ARK server API supported for support of ark plugins?

**Comment by mschnitzer** [comment]
Do you mean mod support? Yes, there's a [section in the README](https://github.com/mschnitzer/ark-survival-ascended-linux-container-image?tab=readme-ov-file#adding-mods).

**Comment by madmediaz2** [comment]
i did not mean mod support, i meant the ark server API. it enabled you to add plugins like discord-cross chat. Here is a link to the ark server API on [https://gameservershub.com/forums/resources/ark-survival-ascended-serverapi-crossplay-supported.683/](Gameservershub). I'm not really familiar with the ARK server API or ark hosting in general. but it wasn't possible enable this on nitrado thats why we switched to a vps on linux. I'am hoping u could tell me if this API is compatible with your docker image.

**Comment by mschnitzer** [comment]
It might work, but it requires some changes to the docker image itself. They want their executable to be executed instead of `ArkAscendedServer.exe`, so to achieve that, this line would need to be adjusted: https://github.com/mschnitzer/ark-survival-ascended-linux-container-image/blob/89dae4791ed8ac2b2e476d4db149f0cf1020536f/root/usr/bin/start_server#L50

As this file is part of the image, I'd need to provide an updated image that allows you to choose the executable that can be launched, otherwise this won't be possible without manual non-persistent changes by the admin inside of the image.

I can save this as a feature request, but it's unlikely I will time to work on it this week.

**Comment by madmediaz2** [comment]
if you tell me to change what and where i could probably do it myself. You think its a hard task?

**Comment by mschnitzer** [comment]
I'm unavailable at the moment. I'm not even sure if it will work on a Linux server. The executale needs to be emulated with the help of Proton, but if that works at all is on another paper. I need some time to really dig into this. Kinda hard to provide a step-by-step guide without knowing whether it works at all and without knowing how the complete setup looks like of that plugin.

**Comment by madmediaz2** [comment]
![image](https://github.com/mschnitzer/ark-survival-ascended-linux-container-image/assets/147717990/08a5b0ba-16fc-419f-ae36-bdf7d23ff19b) i just installed and it seemed to work, i didnt change anything in the config files or anything yet as i'am at work right now. I will do some digging and experimenting about this topic and report back to you. Thanks for the information about which executable to run.

**Comment by mschnitzer** [comment]
OK, I might have misunderstood their installation guide then, just had a brief look. Once you are successful, just let me know what you did and I will test it myself next week.

**Comment by kalewenn** [comment]
> OK, I might have misunderstood their installation guide then, just had a brief look. Once you are successful, just let me know what you did and I will test it myself next week.

You are right, we have to run **AsaApiLoader.exe** instead of _ArkAscendedServer.exe_

**Comment by madmediaz2** [comment]
AsaApiLoader is planning on adding linux support soon, Might be best to wait for that. I didn't try modifying the docker image yet I'm not sure if i will for the coming time as i will be busy.

**Comment by kalewenn** [comment]
> AsaApiLoader is planning on adding linux support soon

They said that at least 9 month ago :D 

> Might be best to wait for that. I didn't try modifying the docker image yet I'm not sure if i will for the coming time as i will be busy.

Ok ! So sad :(

**Comment by mschnitzer** [comment]
I can try something next week and see whether emulating it with Proton works.

I really doubt they will provide native Linux support for their project, when not even ASA has native Linux support, but let's see.

**Comment by kalewenn** [comment]
I don't know if it can help you but i found this : https://gameservershub.com/forums/threads/ark-sa-server-pterodactyl-egg-with-api.5154/#post-46623

**Comment by madmediaz2** [comment]
> I don't know if it can help you but i found this : https://gameservershub.com/forums/threads/ark-sa-server-pterodactyl-egg-with-api.5154/#post-46623

Thanks very much i will forward it to my server manager :)

**Comment by kalewenn** [comment]
> I can try something next week and see whether emulating it with Proton works.

Hello @mschnitzer ,
I just wanted to say that I tried to emulate AsaApiLoader.exe with Proton and it worked !

If you are interested, you have to know that I did it very dirty :
* I ran my standard container and edited my /usr/bin/start_server inside the container, and added a sleep before running ArkAscendedServer.exe
* I copied inside my container the AsaApi files (AsaApiLoader.exe and all the files needed) in Win64 folder
* I restarted my container so my server don't start
* Inside my container, I ran the Proton run manually (note that I updated the Proton version on last 8th release) : 

```
cd /home/gameserver/server-files/ShooterGame/Binaries/Win64
/home/gameserver/Steam/compatibilitytools.d/GE-Proton8-32/proton run ./AsaApiLoader.exe TheIsland_WP?listen?Port...
```

logs:

```
[38;2;000;128;000m[Success]: Detected Server: ArkAscendedServer.exe
[0m[38;2;000;128;000m[Success]: Detected Api: AsaApi.dll
[0m05/13/24 15:48 [API][info] -----------------------------------------------
05/13/24 15:48 [API][info] ARK:SA Api V1.15
05/13/24 15:48 [API][info] Brought to you by ArkServerApi
05/13/24 15:48 [API][info] https://github.com/orgs/ArkServerApi
05/13/24 15:48 [API][info] Loading...

05/13/24 15:48 [API][error] Requested file 'https://cdn.pelayori.com/cache/c09a94bc1...0904aaf.zip?hash=58f4c10...263b2' failed with error: 'Forbidden'
05/13/24 15:48 [API][error] Failed to download the Cache files.
05/13/24 15:48 [API][info] Cache refresh required this will take 10-20 minutes to complete
05/13/24 15:48 [API][info] Dumping structures..
05/13/24 15:50 [API][info] Dumping functions..
05/13/24 16:09 [API][info] Dumping globals..
05/13/24 16:09 [API][info] Successfully read information from PDB

05/13/24 16:09 [API][info] Caching offsets for faster loading next time
05/13/24 16:09 [API][info] Caching bitfields for faster loading next time
05/13/24 16:09 [API][info] Initialized hooks

05/13/24 16:09 [API][info] API was successfully loaded
05/13/24 16:09 [API][info] -----------------------------------------------

[38;2;000;128;000m[Success]: Loader Completed, Reason: Success
[0m05/13/24 16:09 [API][info] UGameEngine::Init was called
05/13/24 16:09 [API][info] Loading plugins..

```

I then had an error because I haven't configured a database yet, but main work is done.

Now it's time to make it generic, configurable... 
If you want I perhaps can help you if you tell me what you really want.
Adding plugins in ASA Servers is a must-have that can bring a lot of downloads on your wonderful repository. Thanks for everything dude <3

**Comment by mschnitzer** [comment]
Sounds good, let me have a look this week. Thanks for testing it!

**Comment by mschnitzer** [comment]
Almost done with the implementation, releasing version `1.4.0` of the container image today or tomorrow.

**Comment by mschnitzer** [comment]
I'm done with the general implementation. I didn't do a release yet of version `1.4.0`, because I'm a little unsatisfied with the size of the container image. I will optimize it before I do a full release.

However you can test it already by pulling the image from the beta branch. Just adjust the `docker-compose.yml` file an change in the `image:` section the `latest` with `beta`, so that it looks like this: `image: "mschnitzer/asa-linux-server:beta"`

Please also add a `tty: true` below the `image:` tag, like so:

```yml
    image: "mschnitzer/asa-linux-server:beta"
    tty: true
```

This is necessary to launch the ASA API plugin.

There's no further adjustments required. Just follow [this guide](https://github.com/mschnitzer/ark-survival-ascended-linux-container-image/blob/dev/README.md#adding-plugins) of the `dev` branch.

Feel free to share your feedback.

**Comment by mschnitzer** [comment]
Nevermind, I was able to reduce the filesize and released 1.4.0:
https://github.com/mschnitzer/ark-survival-ascended-linux-container-image/releases/tag/1.4.0

I'll close this feature. Please reopen if something is not working as expected.
